### PR TITLE
Task-53542: Hidden Space's Activity: Hidden Space ID can be found in code

### PR DESCRIPTION
**Pull request by wassimmhamdi**
Problem: by inspecting the page source code, of a Hidden Space, the space URL is displayed to users who are not in this space Fix: No information related to hidden spaces is displayed for users that are not its members

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 1]
```diff
@@ -52,6 +52,9 @@ export default {
       return `${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/social/spaces/default-image/avatar`;
     },
     url() {
+      if (!this.space || !this.space.isMember) {
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: I think the test is not ok.
We want to remove the link when the space is hidden.
If the space is not hidden and if I'm not member, I should be able to click on the link on request access.

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
My initial comment was not adressed.
Please reread the expected behaviour, and update your fix,
Thanks

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 2]
```diff
@@ -39,8 +39,16 @@ export default {
     spaceId() {
       return this.space && this.space.id;
     },
+    testvisibility(){
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: testVisibility is not a function name easy to read.
Can you change to something like 
canClickOnSpace

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 3]
```diff
@@ -2,8 +2,8 @@
   <a
     :id="id"
     :href="url"
-    :class="!this.space.isMember && 'not-clickable-link hidden-space'"
-    class="text-none space-avatar primary--text activity-head-space-link">
+    :class="testvisibility && 'not-clickable-link hidden-space'"
+    class="text-none space-avatar activity-head-space-link">
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue:6]: why did you remove primary--text ?

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 4]
```diff
@@ -52,6 +60,9 @@ export default {
       return `${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/social/spaces/default-image/avatar`;
     },
     url() {
+      if (this.testvisibility) {
+        return null ;
+      }
       if (!this.groupId) {
         return '#';
       }
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: could we have something like this :
```suggestion
      if (this.testvisibility || !this.groupId) {
        return '#';
      }
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 3 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue:6]: I remove the class primary--text to make the hidden space name be with the grey color

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 2 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: ok I changed it

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 4 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: Yes !

File: webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties [hunk 5]
```diff
@@ -414,7 +414,7 @@ UsersManagement.error.importCSV.uploadedFileIsAlreadyProcessing=File is already
 UsersManagement.error.importCSV.errorReadingUploadedFile=Error occurred while processing uploaded file
 UsersManagement.error.importCSV.fileIsEmpty=Uploaded file is empty
 UsersManagement.error.importCSV.missingUserName=Username is mandatory
-UsersManagement.error.importCSV.badLineFormat=CSV Line for user {0} has a bad format (check column count)
+UsersManagement.error.importCSV.badLineFormat=Line of user {0} does not contain the expected number of columns
```
Comment (rdenarie) [inline webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties]: Why do you have this modification ?

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 3 above)
Comment (rdenarie) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue:6]: and If the link is clickable, does primary--text is used ?

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 6]
```diff
@@ -52,8 +60,8 @@ export default {
       return `${eXo.env.portal.context}/${eXo.env.portal.rest}/v1/social/spaces/default-image/avatar`;
     },
     url() {
-      if (!this.groupId) {
-        return '#';
+      if (this.canClickOnSpace || !this.groupId) {
+        return '#' ;
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]:
```suggestion
        return '#';
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 6 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: resolved

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 3 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue:6]: yes, primary--text is not used only if the space is hidden and not clickable

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue [hunk 7]
```diff
@@ -39,8 +39,16 @@ export default {
     spaceId() {
       return this.space && this.space.id;
     },
+    canClickOnSpace(){
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: the name is KO.
the function is canClickOnSpace, but return false if user can click on the space.
Can you update the function name or the function content

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue (hunk 7 above)
Comment (wassimmhamdi) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/header/ActivityHeadSpace.vue]: I updated the function name

**Review by rdenarie — APPROVED** [review]
